rcl_action/test/rcl_action/test_graph.cpp

```diff
@@ -593,3 +594,75 @@ TEST_F(TestActionGraphMultiNodeFixture, test_action_get_client_names_and_types_b
   ret = rcl_names_and_types_fini(&nat);
   EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
 }
+
+TEST_F(TestActionGraphMultiNodeFixture, get_names_and_types_maybe_fail)
+{
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    rcl_names_and_types_t nat = rcl_get_zero_initialized_names_and_types();
+    rcl_ret_t ret = rcl_action_get_names_and_types(&this->node, &this->allocator, &nat);
+    if (RCL_RET_OK == ret) {
```

Comment:
what happens if this one fails?

Reply:
Here and below cases where init would fail the error state might require to be reset

Reply:
Added logic to finalize a second time for names_and_types. The rcl_action_client one at line 632 can only have its error reset.